src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -327,6 +327,10 @@ private WebHostBuilderContext GetWebHostBuilderContext(HostBuilderContext contex
                 return webHostBuilderContext;
             }
 
+            // Refresh config, it's periodically updated/replaced
+            var webHostContext = (WebHostBuilderContext)contextVal;
+            webHostContext.Configuration = context.Configuration;
+
             return (WebHostBuilderContext)contextVal;
```

Comment:
You could now return `webHostContext` directly.

Reply:
😄 Yes